src/main/scala/nl/knaw/dans/easy/fedora2vault/EasyFedora2vaultApp.scala

```diff
@@ -41,18 +43,26 @@ class EasyFedora2vaultApp(configuration: Configuration) extends DebugEnhancedLog
   private lazy val ldap = new Ldap(ldapContext)
   private val emdUnmarshaller = new EmdUnmarshaller(classOf[EasyMetadataImpl])
 
-  def simpleTransForms(input: File, outputDir: File): Try[FeedBackMessage] = {
-    input.lineIterator.filterNot(_.startsWith("#")).map(datasetId => {
-      simpleTransform(datasetId, outputDir)
-        .doIfFailure { case t =>
-          logger.error(s"$datasetId failed", t)
+  def simpleTransForms(input: File, outputDir: File)(implicit printer: CSVPrinter): Try[FeedBackMessage] = {
+    input.lineIterator.filterNot(_.startsWith("#")).map { datasetId =>
+      val uuid = UUID.randomUUID
+      simpleTransform(outputDir / uuid.toString)(datasetId)
+        .doIfFailure { case t => logger.error(s"$datasetId -> $uuid failed: $t", t) }
+        .recoverWith {
+          case t: FedoraClientException if t.getStatus != 404 => Failure(t)
+          case t => CsvRecord(
+            datasetId, doi = "", depositor = "", SIMPLE, uuid, s"FAILED: $t"
```

Comment:
not sure what `t.toString` is for an exception. Is that only the message, or also the stacktrace? For custom exceptions (case classes in Scala) it will probably look different that exceptions from JDK. Please test and adjust if necessary.

Reply:
it's the qualified class name plus message
![image](https://user-images.githubusercontent.com/10553630/81386871-06589c80-9116-11ea-8a46-08b76e28c4a3.png)